org.eclipse.lsp4e/src/org/eclipse/lsp4e/operations/typeHierarchy/TypeHierarchyViewContentProvider.java

```diff
@@ -125,10 +124,8 @@ private void initialise(final @NonNull IDocument document, final int offset, Tre
 							if (treeViewer != null) {
 								treeViewer.refresh();
 								treeViewer.expandToLevel(2);
-								if (hierarchyItems != null && !hierarchyItems.isEmpty()) {
```

Comment:
you now also removed the `!hierarchyItems.isEmpty()` check. Was that on purpose? `hierarchyItems.get(0)` will fail if the list is empty

Reply:
> Was that on purpose? 

Yes, because `hierarchyItems.get(0)` can't fail since  `hierarchyItems` will be assigned to `p.getSecond()`which will be checked  to be non empty in the if condition (see line 120 and following):
```java
if (!p.getSecond().isEmpty()) {
  hierarchyItems = p.getSecond();
  treeViewer = viewer;
  PlatformUI.getWorkbench().getDisplay().asyncExec(() -> {
	  if (treeViewer != null) {
		  treeViewer.refresh();
		  treeViewer.expandToLevel(2);
		  treeViewer.getControl().setEnabled(true);
		  treeViewer.setSelection(new StructuredSelection(hierarchyItems.get(0)));
	  }
   });
}
```
`hierarchyItems` will be assinged to `Collections.emptyList()` only during init or in `handleRootError`.

Reply:
I added the check again